### PR TITLE
Add XS variants of items

### LINKED ISSUE #47762: Feature Request: XS clothing
# Is your feature request related to a problem? Please describe.
Currently playing on a character with the Tiny mutation. A lot of pre-generated clothing doesn't fit, which is fine. **However, a good deal of the clothing that my character herself can make would not fit.** If she made herself her own mittens, for example, they would be too big no matter what. This doesn't make sense! If I was making my own clothes, I would make them fit me, rather than some mythical average human, and it's frustrating having a significant amount of clothing and armor locked from my access.

# Describe the solution you'd like
**Recipes for XS items for every item that has an XL version -- ideally for every recipe that doesn't fit abnormally small mutated anatomy and that cannot be downsized.** These would mostly be copies of the original item, but weigh slightly less, require slightly less materials, and have slightly lower volume. Even just direct copies with no changes other than fit would be a great first step, and I'm considering making a PR with those available.

# Describe alternatives you've considered
**Making items made by the player automatically fit their anatomy, just like items made by the player are automatically a good fit.** This has the advantage of not cluttering up the crafting menu with duplicate recipes, but it doesn't allow for changes in weight, volume, cost, etc., and it would also likely require changes to the actual code instead of just the json.

**Going through and reinvestigating which items can or can't fit mutated anatomy, and which items can or can't be refitted.** For example, while I can _maybe_ accept that a bandana could possibly be too large if you had like... a _really_ small face, it seems really strange to me that you can't downsize/refit it -- just fold the edges over and sew them up, or even trim the bandana with a pair of scissors. I'm sure this isn't the only item like this. In general it seems to be that the decision whether an item could be upsized/downsized/refitted was made more with abnormally large anatomy in mind, rather than abnormally small.

### PR DESCRIPTION
**Pull request by UmbralReaper**
#### Summary
Content "Adds XS versions of items that cannot be resized"

#### Purpose of change
This is #48158 reborn. 
partially fixes #47762, related to #54047, #53496

#### Describe the solution

Generate XS variants of un-resizable items. Given the relatively automated nature 

<details>
    <summary>TODO</summary>

- [x] Change components to 75%
- [x] Change all `id`s to match the `id` of the XL version.
- [x] Add masks

**Recipes**
- [x] `xl_chainmail_hood`
- [x] `xl_armor_lightplate`
- [x] `xl_chainmail_suit`
- [x] `xl_kevlar_chainmail_suit`
- [x] `jumpsuit_xl`

**Uncraft**
- [x] `xlsocks`
- [x] `xlboxer_shorts`
- [x] `xlboxer_briefs`
- [x] `xlboy_shorts`
- [x] `xlsurvivor_suit`

</details>

#### Describe alternatives you've considered

Not doing this.

#### Testing

None, yet.

#### Additional context

@LeahLuong

### CODE REVIEW

**Comment by UmbralReaper** [comment]
@Maleclypse I'm unsure about how much to change the requirements for each item. Currently I have a flat 50% on components for each recipe, and a flat 75% on price / volume / weight of the resulting item.

**Comment by Maleclypse** [comment]
> @Maleclypse I'm unsure about how much to change the requirements for each item. Currently I have a flat 50% on components for each recipe, and a flat 75% on price / volume / weight of the resulting item.

Probably do a flat 75% across the board so components and volume/weight.  Price would be more like 50% I think, since it would have significantly fewer buyers.

File: data/json/items/armor/arms_armor.json [hunk 1]
```diff
@@ -72,6 +72,16 @@
     "proportional": { "weight": 1.25, "volume": 1.23 },
     "flags": [ "STURDY", "OUTER", "BLOCK_WHILE_WORN", "WATER_FRIENDLY", "OVERSIZE" ]
   },
+  {
+    "id": "armguard_chitin_xs",
+    "type": "ARMOR",
+    "name": { "str": "pair of XS chitin arm guards", "str_pl": "pairs of XS chitin arm guards" },
+    "description": "A pair of arm guards made from the exoskeletons of insects.  Light and durable. Smaller than average.",
```
Comment (Maleclypse) [inline data/json/items/armor/arms_armor.json:79]:
```suggestion
    "description": "A pair of arm guards made from the exoskeletons of insects.  Light and durable.  Smaller than average.",
```

File: data/json/items/armor/arms_armor.json [hunk 2]
```diff
@@ -136,6 +156,16 @@
     "copy-from": "armguard_hard",
     "flags": [ "OUTER", "WATER_FRIENDLY", "BLOCK_WHILE_WORN", "OVERSIZE" ]
   },
+  {
+    "id": "armguard_hard_xs",
+    "type": "ARMOR",
+    "name": { "str": "pair of XS hard arm guards", "str_pl": "pairs of XS hard arm guards" },
+    "description": "A pair of neoprene arm sleeves covered with molded plastic sheaths. Tiny sized.",
```
Comment (Maleclypse) [inline data/json/items/armor/arms_armor.json:163]:
```suggestion
    "description": "A pair of neoprene arm sleeves covered with molded plastic sheaths.  Tiny sized.",
```

File: data/json/items/armor/arms_armor.json [hunk 3]
```diff
@@ -258,6 +308,16 @@
     "copy-from": "armguard_metal",
     "flags": [ "OUTER", "BLOCK_WHILE_WORN", "WATER_FRIENDLY", "OVERSIZE" ]
   },
+  {
+    "id": "armguard_metal_xs",
+    "type": "ARMOR",
+    "name": { "str": "pair of XS metal arm guards", "str_pl": "pairs of XS metal arm guards" },
+    "description": "A pair of arm guards hammered out from metal.  Perfect for the Post-Apocalyptic Warrior look. Smaller than average.",
```
Comment (Maleclypse) [inline data/json/items/armor/arms_armor.json:315]:
```suggestion
    "description": "A pair of arm guards hammered out from metal.  Perfect for the Post-Apocalyptic Warrior look.  Smaller than average.",
```

File: data/json/items/armor/arms_armor.json [hunk 4]
```diff
@@ -459,6 +549,16 @@
     "copy-from": "chainmail_arms",
     "flags": [ "VARSIZE", "STURDY", "OVERSIZE" ]
   },
+  {
+    "id": "chainmail_arms_xs",
+    "type": "ARMOR",
+    "name": { "str": "pair of XS chainmail sleeves", "str_pl": "pairs of XS chainmail sleeves" },
+    "description": "Customized chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves. These are undersized",
```
Comment (Maleclypse) [inline data/json/items/armor/arms_armor.json:556]:
```suggestion
    "description": "Customized chainmail arms.  Each sleeve has straps to connect them with each other.  The lack of fingers makes them less cumbersome and allows them to be used with gloves.  These are undersized",
```

File: data/json/items/armor/helmets.json [hunk 5]
```diff
@@ -194,6 +194,16 @@
     "proportional": { "weight": 1.25, "volume": 1.3, "price": 1.25 },
     "extend": { "flags": [ "OVERSIZE" ] }
   },
+  {
+    "id": "helmet_barbute_xs",
+    "type": "ARMOR",
+    "name": { "str": "XS barbute helm" },
+    "description": "A medieval helmet that provides excellent protection for the head, with a Y shaped opening for the face. This one is goblin sized.",
```
Comment (Maleclypse) [inline data/json/items/armor/helmets.json:201]:
```suggestion
    "description": "A medieval helmet that provides excellent protection for the head, with a Y shaped opening for the face.  This one is goblin sized.",
```

**Comment by UmbralReaper** [comment]
I'm not sure why the tests are failing.

**Pull request merged**